auth-table-based-service/src/main/java/io/stargate/auth/table/AuthnTableBasedService.java

```diff
@@ -226,18 +233,39 @@ protected static boolean checkpw(String password, String hash) {
 
   @Override
   public AuthenticationSubject validateToken(String token) throws UnauthorizedException {
+    // if not there fail fast
     if (Strings.isNullOrEmpty(token)) {
       throw new UnauthorizedException("authorization failed - missing token");
     }
 
+    // otherwise, look in the cache and optionally fetch if missing
+    try {
+      return tokenCache.get(token, () -> getAuthenticationSubject(token));
+    } catch (ExecutionException e) {
+      // properly inspect the cause of the execution exception
+      // and re-throw if UnauthorizedException or RuntimeException
+      // otherwise wrap in the RuntimeException
+      Throwable cause = e.getCause();
+      if (cause instanceof UnauthorizedException) {
+        throw (UnauthorizedException) cause;
+      } else if (cause instanceof RuntimeException) {
```

Comment:
Does the cache wrap everything in `ExecutionException`?

Reply:
Yep and this was confirmed in tests that were failing with 500 instead of unauthorized before I added this..